webapp/src/assets.ts

```diff
@@ -118,4 +118,13 @@ export function assetToGalleryItem(asset: pxt.Asset, imgConv = new pxt.ImageConv
 
 function compareInternalId(a: pxt.Asset, b: pxt.Asset) {
     return a.internalID - b.internalID;
+}
+
+function comparePackage(a: pxt.Asset, b: pxt.Asset) {
+    const aPack = a.meta.package === "device" ? 1 : 0;
```

Comment:
instead of hardcoding device, we could probably detect what the  default package is:

```ts
const blocksprj = pxt.appTarget.bundledpkgs[pxt.BLOCKS_PROJECT_NAME];
            const config = blocksprj?.[pxt.CONFIG_NAME];
            if (config) {
                const pkgConfig: pxt.PackageConfig = JSON.parse(config);
                // cache this and check for the package id in this list
                const dependencies = Object.keys(pkgConfig.dependencies);
            }
```
obviously you'd want to cache the result from this instead of reparsing the config for each comparison
i haven't tested this snippet, so it might need tweaking